src/operations/indexes.ts

```diff
@@ -193,14 +194,46 @@ function makeIndexSpec(
     }
   }
 
-  return { ...options, key };
+  return key;
 }
 
+/**
+ * Receives an index description and returns a modified index description which has had invalid options removed
+ * from the description and has mapped the `version` option to the `v` option.
+ */
+function resolveIndexDescription(
+  description: IndexDescription
+): Omit<ResolvedIndexDescription, 'key'> {
+  const validProvidedOptions = Object.entries({ ...description }).filter(([optionName]) =>
```

Comment:
```suggestion
  const validProvidedOptions = Object.entries(description).filter(([optionName]) =>
```

Isn't the spread redundant? Or did I miss something about the input type?

Reply:
it's a speed-up loop: https://thedailywtf.com/articles/The-Speedup-Loop

Just joking 😄 this change should be safe, it was just copied from the original implementation.